plugins/README.adoc

```diff
@@ -7,3 +7,247 @@ NOTE: Work in Progress...
 * Review Ansible guidelines for https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_best_practices.html[modules] and https://docs.ansible.com/ansible/latest/dev_guide/index.html[development].
 * Use https://pep8.org/[PEP8].
 * File headers and functions should have comments for their intent.
+
+
+== All plugin types should use argspec validation, inbound and outbound
+[%collapsible]
+===
+Explanations:: A plugin's `argument_spec` (or "argspec") defines the supported arguments for a module, as well as their type, defaults and more.
+
+Rationale:: Validating the argspec at the time that it enters and exits the code ensures that these inputs and outputs conform to the plugin's documentation and the user's expectations.
+
+Examples::
+From the https://github.com/ansible-collections/ansible.utils/blob/2.6.0/plugins/filter/ipaddr.py[ipaddr] filter plugin.
```

Comment:
I don't know if we have an example ready, but maybe we can also validate the return payload by comparing it with the `RETURN` defined in the plugin file. That way we can guarantee that the plugin returns _exactly_ what it says it does.

Reply:
@NilashishC +1 for validating the return payload for the module. Maybe this can be in the ansible-core argspec validation.

Reply:
Does it make sense to keep this as a generic recommendation?

Reply:
As we can't provide documentation or user-accessible examples I'm removing this recommendation.  It can be proposed again in the future if the circumstance changes.